src/Browser.php

```diff
@@ -783,6 +852,23 @@ private function requestMayBeStreaming($method, $url, array $headers = array(),
             $body = new ReadableBodyStream($body);
         }
 
+        foreach ($this->defaultHeaders as $key => $value) {
+            if ($headers === array()) {
+                $headers = $this->defaultHeaders;
+                break;
+            }
+
```

Comment:
I don't think this is needed? In this case, I would rather try to reduce cyclomatic complexity:

```suggestion
```